src/lib/NonLinearDutchDecayLib.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {OutputToken, InputToken} from "../base/ReactorStructs.sol";
+import {NonLinearDutchOutput, NonLinearDutchInput, NonLinearDecay} from "../lib/NonLinearDutchOrderLib.sol";
+import {Util} from "./Util.sol";
+import {FixedPointMathLib} from "solmate/src/utils/FixedPointMathLib.sol";
+
+/// @notice helpers for handling dutch order objects
+library NonLinearDutchDecayLib {
+    using FixedPointMathLib for uint256;
+
+    /// @notice locates the current position on the curve and calculates the decay
+    function decay(NonLinearDecay memory curve, uint256 startAmount, uint256 decayStartBlock)
+        internal
+        view
+        returns (uint256 decayedAmount)
+    {
+        // handle current block before decay or no decay
+        if (decayStartBlock >= block.number) {
+            return startAmount;
+        }
+        uint256 blockDelta = block.number - decayStartBlock;
```

Comment:
readability nit: consider separating out the process of finding the block/amount bounds from the process of calculating decay i.e. a helper `getBlockBounds(NonLinearDecay) returns (startBlock, endBlock, startAmount, endAmount)`?